control-plane/pkg/reconciler/brokeryolo/controller.go

```diff
@@ -0,0 +1,143 @@
+			KubeClient:                  kubeclient.Get(ctx),
+			PodLister:                   podinformer.Get(ctx).Lister(),
+			SecretLister:                secretinformer.Get(ctx).Lister(),
+			DataPlaneConfigMapNamespace: env.DataPlaneConfigMapNamespace,
+			DataPlaneConfigMapName:      env.DataPlaneConfigMapName,
+			DataPlaneConfigFormat:       env.DataPlaneConfigFormat,
+			DataPlaneNamespace:          env.SystemNamespace,
+			DispatcherLabel:             base.BrokerDispatcherLabel,
+			ReceiverLabel:               base.BrokerReceiverLabel,
+		},
+		NewKafkaClusterAdminClient: sarama.NewClusterAdmin,
+		ConfigMapLister:            configmapInformer.Lister(),
+		Env:                        env,
+		BaseDataPlaneManifest:      manifest,
+	}
+
+	impl := brokerreconciler.NewImpl(ctx, reconciler, kafka.BrokerClassYolo, func(impl *controller.Impl) controller.Options {
+		return controller.Options{PromoteFilterFunc: kafka.BrokerYoloClassFilter()}
+	})
+
+	reconciler.Resolver = resolver.NewURIResolverFromTracker(ctx, impl.Tracker)
+	reconciler.IPsLister = prober.NewIPListerWithMapping()
+	reconciler.Prober = prober.NewAsync(ctx, http.DefaultClient, env.IngressPodPort, reconciler.IPsLister.List, impl.EnqueueKey)
+
+	brokerInformer := brokerinformer.Get(ctx)
+
+	brokerInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: kafka.BrokerYoloClassFilter(),
+		Handler:    controller.HandleAll(impl.Enqueue),
+	})
+
+	reconciler.SecretTracker = impl.Tracker
+	secretinformer.Get(ctx).Informer().AddEventHandler(controller.HandleAll(reconciler.SecretTracker.OnChanged))
+
+	reconciler.ConfigMapTracker = impl.Tracker
+	configmapinformer.Get(ctx).Informer().AddEventHandler(controller.HandleAll(
+		// Call the tracker's OnChanged method, but we've seen the objects
+		// coming through this path missing TypeMeta, so ensure it is properly
+		// populated.
+		controller.EnsureTypeMeta(
+			reconciler.ConfigMapTracker.OnChanged,
+			corev1.SchemeGroupVersion.WithKind("ConfigMap"),
+		),
+	))
+
+	brokerInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: kafka.BrokerYoloClassFilter(),
+		Handler: cache.ResourceEventHandlerFuncs{
+			DeleteFunc: reconciler.OnDeleteObserver,
+		},
+	})
+
+	return impl
+}
+
+func getBaseDataPlaneManifest(client mf.Client) (mf.Manifest, error) {
+	kodatapath := os.Getenv("KO_DATA_PATH")
+	if kodatapath == "" {
+		return mf.Manifest{}, fmt.Errorf("KO_DATA_PATH is empty")
+	}
+	// TODO: create constant!
+	dataplaneManifestPath := kodatapath + "/dataplane/config/broker"
+	manifest, err := mf.ManifestFrom(mf.Path(dataplaneManifestPath), mf.UseClient(client))
+	if err != nil {
+		return mf.Manifest{}, fmt.Errorf("unable to load dataplane manifest from path '%s': %v", dataplaneManifestPath, err)
+	}
+
+	// TODO: need to find a better way!
+	// exclude the cluster scoped resources
+	manifest = manifest.Filter(mf.Not(mf.Any(mf.ByKind("ClusterRole"), mf.ByKind("ClusterRoleBinding"))))
```

Comment:
We would need to convert these resources to Role and RoleBinding

Reply:
I wrote some notes about these in the PR desc. Because of this, and additional problems, I am not reusing the manifests in `dataplane/config` for creating the dynamic dataplane.

This line is not really necessary anymore because there are no ClusterRoles nor ClusterRoleBindings in the dynamic dataplane manifest.

Gonna remove this line